includes/Classifai/Admin/SavePostHandler.php

```diff
@@ -192,10 +192,18 @@ public function synthesize_speech( $post_id ) {
 			);
 		}
 
-		$normalizer   = new Normalizer();
-		$settings     = \Classifai\get_plugin_settings( 'language_processing', TextToSpeech::FEATURE_NAME );
-		$post         = get_post( $post_id );
-		$post_content = $normalizer->normalize_content( $post->post_content, $post->post_title, $post_id );
+		$normalizer          = new Normalizer();
+		$settings            = \Classifai\get_plugin_settings( 'language_processing', TextToSpeech::FEATURE_NAME );
+		$post                = get_post( $post_id );
+		$post_content        = $normalizer->normalize_content( $post->post_content, $post->post_title, $post_id );
+		$content_hash        = get_post_meta( $post_id, TextToSpeech::AUDIO_HASH_KEY, true );
+		$saved_attachment_id = (int) get_post_meta( $post_id, TextToSpeech::AUDIO_ID_KEY, true );
+		if ( ! empty( $content_hash ) ) {
```

Comment:
Seems like we may want a check here to ensure the `$saved_attachment_id` actually exists, in case this was deleted somehow. We could also combine this `if` statement into a single line:
```suggestion
		if ( ! empty( $content_hash ) && md5( $post_content ) === $content_hash && $saved_attachment_id  ) {
```

Reply:
I would also maybe add a comment above this `if` statement describing what we're doing. Something like:
```php
// Don't regenerate the audio file it it already exists and the content hasn't changed.
```